src/main/scala/com/wavesplatform/state2/reader/CompositeStateReader.scala

```diff
@@ -33,23 +33,20 @@ class CompositeStateReader(inner: StateReader, blockDiff: BlockDiff) extends Sta
   override def slotAddress(id: Int): Option[String] =
     txDiff.slotids.get(id) match {
       case None => inner.slotAddress(id).orElse(None)
-      case add if add.get == "" => None
-      case add => add
+      case Some(None) => None
+      case Some(add) => add
     }
 
-  override def effectiveSlotAddressSize: Int = inner.effectiveSlotAddressSize + txDiff.slotNum
-
-  override def addressToSlotID(add: String): Option[Int] = {
-    inner.addressToSlotID(add) match {
-      case None => txDiff.slotids.filter(_._2 == add).keys.headOption
-      case id => txDiff.slotids.filter(_._1 == id.get).values.headOption match {
-        case None => id
-        case adx if adx.get == add => id
-        case _ => None
-      }
+  override def addressSlot(add: String): Option[Int] = {
+    txDiff.addToSlot.get(add) match {
+      case None => inner.addressSlot(add)
+      case Some(None) => None
+      case Some(id) => id
```

Comment:
@zl730 a little bit confused here~ plz double check it.

Reply:
I think the code in last commit is enough.
Map.get returns Option, "None" is for value not found in map, "Some(None)" for the value None put in map (```Map(sender.toAddress.address -> None)```). 
They are in different layer.
Or you may consider about using ```getOrElse()```
 ```scala
txDiff.addToSlot.get(add).getOrElse(inner.addressSlot(add))
```

Reply:
thx, i see.
`val z: Map[Int, Option[Int]] = Map(1->None)`
`z(1).get` is not defined
but `z.get(1)` is `Option[Option[Int]]` (`Some(None)`), 

better to use `getOrElse()`